conformance/tests/httproute-disallowed-kind.go

```diff
@@ -43,6 +43,7 @@ var HTTPRouteDisallowedKind = suite.ConformanceTest{
 
 		routeNN := types.NamespacedName{Name: "disallowed-kind", Namespace: "gateway-conformance-infra"}
 		gwNN := types.NamespacedName{Name: "tlsroutes-only", Namespace: "gateway-conformance-infra"}
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
```

Comment:
should this be ResolvedRefs = True?

```
	// Possible reasons for this condition to be false are:
	//
	// * "RefNotPermitted"
	// * "InvalidKind"
	// * "BackendNotFound"
```

False and reason `InvalidKind` seems more appropriate here?

Reply:
i guess ResolvedRefs in this case is just for backendRefs?

Reply:
hm i guess in Contour's case we stop processing a Route once it has an invalid parent reference so we don't continue on to validate the rest of its references

Reply:
I would maybe push back a little on this line in the test, just because this is an error case so prescribing how an implementation handles error cases like this (that you still have to process the rest of the resource) is a little odd

Reply:
I think the original intention here was that:

1. the reference itself was permitted
2. the kind was valid
3. the backend was found

These three aspects currently comprise the list of things that would make the condition false:

```go
        // This condition indicates whether the controller was able to resolve all
        // the object references for the Route.
        //
        // Possible reasons for this condition to be true are:
        //
        // * "ResolvedRefs"
        //
        // Possible reasons for this condition to be false are:
        //
        // * "RefNotPermitted"
        // * "InvalidKind"
        // * "BackendNotFound"
        //
        // Controllers may raise this condition with other reasons,
        // but should prefer to use the reasons listed above to improve
        // interoperability.
        RouteConditionResolvedRefs RouteConditionType = "ResolvedRefs"
```

But I can see how saying the references are resolved, but the route is not accepted feels a little bit odd.

After pondering it some more I'm kinda leaning towards sticking with this as it seems pretty reasonable to inform end-users of a situation where "yes I can resolve the references you provided and they are allowed, but I can't accept the route because there's no valid listener". :thinking: 

I'm open to being persuaded however, let me know your thoughts?

Reply:
yeah my bad i saw kind in the test name and immediately thought this line was referencing the "parent" kind but it isn't, it's referencing the backendrefs kinds, so that part of my comment can be ignored

this change https://github.com/kubernetes-sigs/gateway-api/pull/1668 seems to have added this line everywhere but really seems like it should be part of the `GatewayAndHTTPRoutesMustBeAccepted` helper (or something similar) since it is part of validating the base set of assumptions for a happy-path test?

making an assertion about ResolvedRefs in every test feels a *little* off, esp. for error cases where IMO we want to test some specific behavior but don't necessarily care about other things

similar to this test `HTTPRouteInvalidParentRefNotMatchingListenerPort` does not check for `ResolvedRefs`, just the `Accepted` condition

in Contour's case, we don't bother with processing the rest of the Route if it can't actually result in any traffic due to not attaching to a Gateway properly

regardless of what we end up going with, some agreed upon negative/error case handling would be a good outcome (e.g. the full set of conditions should still be present etc.) since we have some inconsistencies here

Reply:
Are you suggesting we should drop the #1668 changes for this release so we can follow up, or do you think we can move forward?

Reply:
@mlavacca interested to get your thoughts here? :thinking:

Reply:
AFAIK, the `Accepted` and `ResolvedRefs` conditions are unrelated, a Route can be accepted/not accepted and can have the `ResolvedRefs` condition either set to true or false, depending on the outcome of the references. For this reason, moving it into the `GatewayAndHTTPRoutesMustBeAccepted` is misleading in my opinion.  

> similar to this test HTTPRouteInvalidParentRefNotMatchingListenerPort does not check for ResolvedRefs, just the Accepted condition

It looked like The ResolvedRefs condition was checked [here](https://github.com/kubernetes-sigs/gateway-api/blob/26080e900c1bf159e36b9d70c2ada5ae8ecf2838/conformance/tests/httproute-invalid-parentref-not-matching-listener-port.go#L45-L51), but the name was wrong, it actually checks the accepted condition. I prepared [this PR](https://github.com/kubernetes-sigs/gateway-api/pull/1819) to fix that and add the `resolvedRefs` check there as well as in the `httproute-invalid-cross-namespace-parent-ref` test.

Reply:
As I understand it abdc042057572a1a85919629a7a3a564cb367b9a should resolve this issue as it adds the missing consistency. @sunjayBhatia if you feel it _doesn't_ resolve it, please unresolve the thread and let us know what prescriptive actions you feel need to be taken next.

Putting this PR back on hold to ensure we resolve @sunjayBhatia's needs and stake in this.

/hold

Reply:
Sounds good to me. Do you mind making an issue for this as a follow-up or would you prefer me to?